interface/client/lib/helpers/helperFunctions.js

```diff
@@ -50,6 +50,28 @@ Helpers.formatUrl = function(url){
 };
 
 /**
+Sanatizes URLs to prevent phishing and XSS attacks
+
+@method sanatizeUrl
+@param {String} url
+**/
+Helpers.sanatizeUrl = function(url){
+    var parser;
+
+    if(_.isString(url)) {
+        parser = document.createElement('a');
+        parser.href = url;
+    } else if(_.isObject(url) && url.protocol) {
+        parser = url;
+    }
+
+    parser.protocol = parser.protocol.replace(/^(javascript:)?(data:)?(:)?/i, 'http:');
```

Comment:
I tested you version and they didn't cover everything.
I found the best way which is here: Here's a link to share:
http://martin.swende.se/tools/inject_it.js/#aW5pdEFwcCh3aW5kb3cpO2FkZENvZGVNaXJyb3IodW5lc2NhcGUoJyUwQSUwQS8vJTIwYmVzdCUwQXNhbmF0aXplVXJsJTIwJTNEJTIwZnVuY3Rpb24lMjh1cmwlMjklN0IlMEElMjAlMjAlMjAlMjB1cmwlMjAlM0QlMjBTdHJpbmclMjh1cmwlMjklM0IlMEElMEElMjAlMjAlMjAlMjB1cmwlMjAlM0QlMjB1cmwucmVwbGFjZSUyOC8lNUIlNUN0JTVDbiU1Q3IlNUNzJTVEKy9nJTJDJTIwJTI3JTI3JTI5JTNCJTBBJTIwJTIwdXJsJTIwJTNEJTIwdXJsLnJlcGxhY2UlMjgvJTVFJTI4JTI4JTNGJTNBamF2YXNjcmlwdCUyOSUzRiUyOCUzRiUzQWRhdGElMjklM0YlNUIlM0ElNUMvJTVDLyU1RCU3QjElMkMzJTdEJTI5L2klMkMlMjAlMjdodHRwJTNBLy8lMjclMjklM0IlMEElMEElMjAlMjAlMjAlMjByZXR1cm4lMjB1cmwlM0IlMEElMjAlN0QlM0IlMEElMEEvKiUwQXNhbmF0aXplVXJsJTIwJTNEJTIwZnVuY3Rpb24lMjh1cmwlMjklN0IlMEElMjAlMjAlMjAlMjB2YXIlMjBwYXJzZXIlMjAlM0QlMjBkb2N1bWVudC5jcmVhdGVFbGVtZW50JTI4JTI3YSUyNyUyOSUzQiUwQSUyMCUyMCUyMCUyMHBhcnNlci5ocmVmJTIwJTNEJTIwJTIyJTIyK3VybCUzQiUyMC8vU3RyaW5naWZ5JTIwVVJMJTBBJTIwJTA5JTBBJTIwJTIwJTIwJTIwaWYlMjglMjBwYXJzZXIucHJvdG9jb2wlMjAlM0QlM0QlMjAlMjJqYXZhc2NyaXB0JTNBJTIyJTIwJTdDJTdDJTBBJTIwJTIwJTIwJTIwJTIwJTIwJTIwcGFyc2VyLnByb3RvY29sJTIwJTNEJTNEJTIwJTIyZGF0YSUzQSUyMiUyOSUwQSUyMCUyMCUyMCUyMCU3QiUwQSUyMCUyMCUyMCUyMCUyMCUyMHBhcnNlci5wcm90b2NvbCUyMCUzRCUyMCUyMmh0dHBzJTNBLy8lMjIlMEElMjAlMjAlMjAlMjAlN0QlMEElMjAlMjAlMjAlMjAlMjByZXR1cm4lMjBwYXJzZXIudG9TdHJpbmclMjglMjklM0IlMEElMjAlN0QlM0IlMEElMjAlMEElMjBzYW5hdGl6ZVVybCUyMCUzRCUyMGZ1bmN0aW9uJTI4dXJsJTI5JTdCJTBBJTIwJTIwJTIwJTIwdmFyJTIwcGFyc2VyJTIwJTNEJTIwZG9jdW1lbnQuY3JlYXRlRWxlbWVudCUyOCUyN2ElMjclMjklM0IlMEElMjAlMjAlMjAlMjBwYXJzZXIuaHJlZiUyMCUzRCUyMCUyMiUyMit1cmwlM0IlMjAvL1N0cmluZ2lmeSUyMFVSTCUwQSUyMCUwOSUwQSUyMCUyMCUyMCUyMGlmJTI4JTIwcGFyc2VyLnByb3RvY29sJTIwJTNEJTNEJTIwJTIyamF2YXNjcmlwdCUzQSUyMiUyMCU3QyU3QyUwQSUyMCUyMCUyMCUyMCUyMCUyMCUyMHBhcnNlci5wcm90b2NvbCUyMCUzRCUzRCUyMCUyMmRhdGElM0ElMjIlMjklMEElMjAlMjAlMjAlMjAlN0IlMEElMjAlMjAlMjAlMjAlMjAlMjBwYXJzZXIucHJvdG9jb2wlMjAlM0QlMjAlMjJodHRwcyUzQS8vJTIyJTBBJTIwJTIwJTIwJTIwJTdEJTBBJTIwJTIwJTIwJTIwJTIwcmV0dXJuJTIwcGFyc2VyLnRvU3RyaW5nJTI4JTI5JTNCJTBBJTIwJTdEJTNCJTBBJTIwJTBBJTIwKi8lMEElMEFzaG93JTI4c2FuYXRpemVVcmwlMjglMjJqYXZhcyU1Q3RjcmlwVCUzQWFsZXJ0JTI4MSUyOSUyMiUyOSUyOSUwQXNob3clMjglMjItLS0tJTIyJTI5JTBBc2hvdyUyOHNhbmF0aXplVXJsJTI4JTIyaHR0cCUzQS8vYWxlcnQlMjgxJTI5JTIyJTI5JTI5JTBBc2hvdyUyOCUyMi0tLS0lMjIlMjklMEFzaG93JTI4c2FuYXRpemVVcmwlMjglMjJodHRwcyUzQS8vdGVzdC5jb20lMjIlMjklMjklMEFzaG93JTI4JTIyLS0tLSUyMiUyOSUwQXNob3clMjhzYW5hdGl6ZVVybCUyOCUyMmphdmFzJTVDdGNyaXBUJTNBLy8lMjUwQWFsZXJ0JTI4MSUyOSUyMiUyOSUyOSUwQXNob3clMjglMjItLS0tJTIyJTI5JTBBc2hvdyUyOHNhbmF0aXplVXJsJTI4JTIyZmlsZSUzQS8vdGVzdC5jb20lMjIlMjklMjklMEFzaG93JTI4JTIyLS0tLSUyMiUyOSUwQXNob3clMjhzYW5hdGl6ZVVybCUyOCUyMmZ0cCUzQS8vdGVzdC5jb20lMjIlMjklMjklMEFzaG93JTI4JTIyLS0tLSUyMiUyOSUwQXNob3clMjhzYW5hdGl6ZVVybCUyOCUyMi8vdGVzdC5jb20lMjIlMjklMjknKSk7DQp1cGRhdGVPdXRwdXQoKTsNCg==

```js
sanatizeUrl = function(url){
// make sure location objects are string too
    url = String(url);

// remove whitespaces and tabs
    url = url.replace(/[\t\n\r\s]+/g, '');
// replace javascript:// data:// and  //
  url = url.replace(/^((?:javascript)?(?:data)?[:\/\/]{1,3})/i, 'http://');

    return url;
 };
```